db/healthcheck.js

```diff
@@ -0,0 +1,44 @@
+import { Client } from '@opensearch-project/opensearch';
+import { URL } from 'url';
+
+/**
+ * Checks the health of the OpenSearch cluster at the given proxy URL.
+ * Retries until healthy or timeout is reached.
+ *
+ * @param {string} proxyUrl - The OpenSearch proxy/base URL.
+ * @param {number} timeout - Maximum seconds to wait for health.
+ * @param {number} interval - Seconds between retries.
+ * @returns {Promise<boolean>} True if healthy, False otherwise.
+ */
+export async function checkOpenSearchHealth(proxyUrl, timeout = 10, interval = 1000) {
+    console.info(`OpenSearch ${proxyUrl} health check started`);
+    const parsed = new URL(proxyUrl);
+    const client = new Client({
+        node: proxyUrl,
+        ssl: {
+            rejectUnauthorized: false
+        }
```

Comment:
Add `requestTimeout: 2000` to the parameters. It'll stop any given loop from potentially hanging if client.cluster.health() just never returns. There is no default timeout, it needs to be specified.